src/mds/mdstypes.h

```diff
@@ -595,6 +595,8 @@ struct inode_t {
   quota_info_t quota;
 
   mds_rank_t export_pin = MDS_RANK_NONE;
+  uint64_t expected_files = 0;
+  uint32_t expected_file_bits = 0;
```

Comment:
I think only need to store one of these into inode_t.  just 'expected_files'

Reply:
It's just a trade-off between storing an extra byte vs recomputing the bits every time we call get_expected_file_bits().  I made it a function so we could change this easily if we would rather save the space.

Reply:
how about add one of them into CInode instead of inode_t. calculate the one in CInode when loading inode.

Reply:
Hrm, that might work.  The thing I'm a little concerned about is the get_val() call in the hot path.  If we never cared about reading the expected_files back, we could just store the bits, and we could probably shrink it to a uint8_t for only a single byte of overhead in the inode_t.